jarvis_files/jarvis_cmd/src/jarvis/launch.py

```diff
@@ -0,0 +1,93 @@
+import configparser
+import os
+import sys
+import shutil
+import time
+import subprocess
+
+from .build import build_dir, build_deps
+
+# Function that parses launch command
+def launch_dir(ctx, d, opts, ssh):
+    if d == "percep":
+        launch_perception(ctx, opts, ssh)
+    if d == "nav":
+        launch_navigation(ctx, opts, ssh)
+    if d == "loc":
+        launch_localization(ctx, opts, ssh)
+    if d == "auton":
+        launch_auton(ctx, opts, ssh)  
+    return
+
+def get_process_id(name):
+
+    child = subprocess.Popen(['pgrep', '-f', name], stdout=subprocess.PIPE, shell=False)
+    response = child.communicate()[0]
+    return [int(pid) for pid in response.split()]
+
+def wait_for_click():
+    pid = get_process_id("xdotool selectwindow")
+    while pid:
+        pid = get_process_id("xdotool selectwindow")
+        time.sleep(.1)
+    return 
+
+def genTerminalLaunchCommand(script_address, ssh):
+    command = "gnome-terminal -- bash -c './"
+    command = command + script_address
+
+    if ssh:
+        return command + " ssh; $SHELL'"
+    else:
+        return command + "; $SHELL'"
+
+# Functions that build and execute auton subteam code
+def launch_perception(ctx, opts, ssh):
+    percep = 'jetson/percep'
+    use_linter = 'True'
+    workspace_relative_address = "jarvis_files/launchScripts/percep"
+    
+    build_dir(ctx, percep, use_linter, opts)
+
+    ctx.run( genTerminalLaunchCommand(workspace_relative_address, ssh) )
+    
+    wait_for_click()
+    return
+
+def launch_navigation(ctx, opts, ssh):
+    nav = 'jetson/nav'
+    use_linter = 'True'
+    workspace_relative_address = "jarvis_files/launchScripts/nav"
+    
+    build_dir(ctx, nav, use_linter, opts)
+    
+    ctx.run( genTerminalLaunchCommand(workspace_relative_address, ssh) )
+
+    wait_for_click()
+    return
+
+def launch_localization(ctx, opts, ssh):
+    gps = 'jetson/gps'
```

Comment:
do names like jetson_gps so that we don't get confused between packages. For example, jetson_gps vs base_station_gps (which are totally different!)

Reply:
Will do